lib/Scheduler/sched_region.cpp

```diff
@@ -164,319 +156,302 @@ FUNC_RESULT SchedRegion::FindOptimalSchedule( //TODO: CHIPPIE: Add helper functi
-  #endif
+#endif
       }
     } else if (rgnTimeout == 0) {
-       Logger::Info(
-           "Bypassing optimal scheduling due to zero time limit with cost %d",
-           bestCost_);
+      Logger::Info(
+          "Bypassing optimal scheduling due to zero time limit with cost %d",
+          bestCost_);
     } else {
-       Logger::Info("The initial schedule of length %d and cost %d is optimal.",
+      Logger::Info("The initial schedule of length %d and cost %d is optimal.",
                    bestSchedLngth_, bestCost_);
     }
 
     if (rgnTimeout != 0) {
-      bool optimalSchedule = initialScheduleOptimal || (rslt == RES_SUCCESS);
-       Logger::Info("Best schedule for DAG %s has cost %d and length %d. The "
+      bool optimalSchedule = isLstOptml || (rslt == RES_SUCCESS);
+      Logger::Info("Best schedule for DAG %s has cost %d and length %d. The "
                    "schedule is %s",
                    dataDepGraph_->GetDagID(), bestCost_, bestSchedLngth_,
                    optimalSchedule ? "optimal" : "not optimal");
     }
-  }
-
-  //TODO: CHIPPIE: Everything hereafter, should it run for _all_ of the schedulers? Regardless of which ones are enabled?
-  //I.e. are the following blocks of code BB specific? (Doesn't look it...)
 
 #ifdef IS_DEBUG_PRINT_PERP_AT_EACH_STEP
-  Logger::Info("Printing PERP at each step in the schedule.");
+    Logger::Info("Printing PERP at each step in the schedule.");
 
-  int costSum = 0;
-  for (int i = 0; i < dataDepGraph_->GetInstCnt(); ++i) {
-    Logger::Info("Cycle: %lu Cost: %lu", i, bestSched_->GetSpillCost(i));
-    costSum += bestSched_->GetSpillCost(i);
-  }
-  Logger::Info("Cost Sum: %lu", costSum);
+    int costSum = 0;
+    for (int i = 0; i < dataDepGraph_->GetInstCnt(); ++i) {
+      Logger::Info("Cycle: %lu Cost: %lu", i, bestSched_->GetSpillCost(i));
+      costSum += bestSched_->GetSpillCost(i);
+    }
+    Logger::Info("Cost Sum: %lu", costSum);
 #endif
 
-  if (SchedulerOptions::getInstance().GetString(
-          "SIMULATE_REGISTER_ALLOCATION") != "NO") {
-    //#ifdef IS_DEBUG
-    RegAlloc_(bestSched, lstSched);
-    //#endif
+    if (SchedulerOptions::getInstance().GetString(
+            "SIMULATE_REGISTER_ALLOCATION") != "NO") {
+      //#ifdef IS_DEBUG
+      RegAlloc_(bestSched, InitialSchedule);
+      //#endif
+    }
+
+    enumTime = Utilities::GetProcessorTime() - enumStart;
+    stats::enumerationTime.Record(enumTime);
   }
 
-  enumTime = Utilities::GetProcessorTime() - enumStart;
-  stats::enumerationTime.Record(enumTime);
+  // Step 5: Run ACO if schedule from enumerator is not optimal
+  if (bestCost_ != 0 && AcoAfterEnum) {
+    Logger::Info("Final cost is not optimal, running ACO.");
+    InstSchedule *AcoAfterEnumSchedule =
+        new InstSchedule(machMdl_, dataDepGraph_, vrfySched_);
+    if (AcoAfterEnumSchedule == NULL)
+      Logger::Fatal("Out of memory");
+
+    FUNC_RESULT acoRslt = runACO(AcoAfterEnumSchedule, bestSched);
```

Comment:
If BB failed to get an optimal schedule (or BB didn't run but AcoBeforeBB did), can we pass their schedules as initial schedules here?

Reply:
The best schedule found from BB or AcoBeforeBB is passed as bestSched.

Reply:
Ah, I missed that the initial schedule is being passed into runACO. My bad.